pkg/manifests/config.go

```diff
@@ -164,6 +164,8 @@ type RemoteWriteSpec struct {
 	QueueConfig *monv1.QueueConfig `json:"queueConfig,omitempty"`
 	// MetadataConfig configures the sending of series metadata to remote storage.
 	MetadataConfig *monv1.MetadataConfig `json:"metadataConfig,omitempty"`
+	// OAuth2 configures OAuth2 authentication for remote write.
+	OAuth2 *monv1.OAuth2 `json:"oauth2,omitempty"`
```

Comment:
@raptorsun for `monv1.OAuth2`, checked in https://github.com/openshift/cluster-monitoring-operator/blob/release-4.11/vendor/github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1/types.go#L1171-L1183
the `OAuth2 struct` do not have `tls_config` field as listed in
https://prometheus.io/docs/prometheus/latest/configuration/configuration/#oauth2

Reply:
That's a good point. I'm going to submit a PR to add this field to the upstream Prometheus Operator 👍
In OAuth2 the necessary configurations are Client ID, Client Secret and TokenURL. 
TLSConfig sets TLS for requesting token from TokenURL, which is not required on all TLS configs, so this still works on most systems. We can live without that field :)